src/dune_action_plugin/dune_action_plugin.mli

```diff
@@ -0,0 +1,81 @@
+(** Applicative and monadic interface for declaring dependencies.
+
+    This module is intended to be used as a interface for declaring
+    dependencies of a computation. Dependencies can be declared dynamically -
+    the list of dependencies can be depend on previous dependencies.
+
+    Note: Monadic "bind" is provided, but it can be very costly. It's called
```

Comment:
Could you be a bit more specific about "very costly"? It's useful to know what exactly is involved in handling a bind appropriately. Is the overhead constant, or linear, i.e. leading to quadratic complexity of a linear chain of binds?

Reply:
Added comment.